src/Common/MoistThermodynamics/profiles.jl

```diff
@@ -0,0 +1,284 @@
+    param_set::APS,
+    z::FT,
+) where {FT <: AbstractFloat}
+    T_surface = ref_state.T_surface
+    T_min = ref_state.T_min
+
+    _grav::FT = grav(param_set)
+    _cp_d::FT = cp_d(param_set)
+    _R_d::FT = R_d(param_set)
+    _MSLP::FT = MSLP(param_set)
+    Γ = _grav / _cp_d
+    z_tropopause = (T_surface - T_min) / Γ
+    H_min = _R_d * T_min / _grav
+    T = max(T_surface - Γ * z, T_min)
+    p = _MSLP * (T / T_surface)^(_grav / (_R_d * Γ))
+    T == T_min && (p = p * exp(-(z - z_tropopause) / H_min))
+    ρ = p / (_R_d * T)
+    return T, p, ρ
+end
+
+"""
+    VirtualTemperature{FT}
+
+A virtual temperature profile that decays smoothly with height `z`, dropping by a specified temperature difference `ΔTv` over a height scale `H_t`.
+
+```math
+Tv(z) = \\max(Tv{\\text{surface}} − ΔTv \\tanh(z/H_{\\text{t}})
+```
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct VirtualTemperature{FT} <: AbstractReferenceState{FT}
+    "Virtual surface at the temperature"
+    T_virt_surf::FT
+    "Temperature drop from surface to top of atmosphere (``ΔT = T_surface - T_TOA``)"
+    ΔT::FT
+    "Height scale over which temperature drops"
+    H_t::FT
+end
+
+function (ref_state::VirtualTemperature)(
+    param_set::APS,
+    z::FT,
+) where {FT <: AbstractFloat}
+    T_virt_surf = ref_state.T_virt_surf
+    ΔT = ref_state.ΔT
+    H_t = ref_state.H_t
+
+    _grav::FT = grav(param_set)
+    _R_d::FT = R_d(param_set)
+    _MSLP::FT = MSLP(param_set)
+
+    # Temperature
+    T = T_virt_surf - ΔT*tanh(z/H_t)
+
+    # Pressure
+    ΔT′ = ΔT/T_virt_surf
+    H_sfc = _R_d*T_virt_surf/_grav
+    num = z + H_t*ΔT′*log(cosh(z/H_t) - atanh(ΔT′))
+    denom = H_sfc*(1 - ΔT′^2)
+    exp_factor = num/denom
+    p = _MSLP * exp(-exp_factor)
+
+    # Density
+    ρ = p / (_R_d * T)
+    return T, p, ρ
+end
+
+"""
+    IsothermalProfile{FT} <: TemperatureProfile
+
+A uniform temperature profile.
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct IsothermalProfile{FT} <: AbstractReferenceState{FT}
+    "temperature (K)"
+    T::FT
+end
+
+function (profile::IsothermalProfile)(
+    param_set,
+    e_pot::FT,
+) where {FT<:AbstractFloat}
+    _R_d::FT = R_d(param_set)
+    _MSLP::FT = MSLP(param_set)
+
+    # Pressure
+    p = _MSLP * exp(-e_pot / (_R_d * profile.T))
+
+    # Density
+    ρ = p / (_R_d * T)
+    return profile.T, p, ρ
+end
+
+"""
+    DryAdiabaticProfile{F} <: AbstractReferenceState
+
+
+A temperature profile that has uniform potential temperature θ until it
+reaches the minimum specified temperature `T_min`
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct DryAdiabaticProfile{FT} <: AbstractReferenceState{FT}
+    "minimum temperature (K)"
+    T_min::FT
+    "potential temperature (K)"
+    θ::FT
+end
+
+function (profile::DryAdiabaticProfile)(
+    param_set,
+    z::FT,
+) where {FT<:AbstractFloat}
+    _cp_d::FT = cp_d(param_set)
+    _grav::FT = grav(param_set)
+    temperature_profile = LinearTemperatureProfile(
```

Comment:
This needs a standalone  specification (see design docs) if the `LinearTemperatureProfile` is removed.